### ISSUE TITLE
Virtual interface in netplan

### ISSUE THREAD

**Issue by alicera** [webmin/webmin#1269]
https://askubuntu.com/questions/990825/virtual-interface-in-netplan

the interface have the error for Virtual interface
This is static ip address setting.It has the Ethernet (Virtual).
But if it is dynamic ip address setting. It hasnt the Ethernet (Virtual). 
![image](https://user-images.githubusercontent.com/59388989/88355019-839ac180-cd95-11ea-910c-62baa0408f69.png)

**Comment by iliarostovtsev** [comment]
Hi,

Latest Webmin 1.953 mustn't have an issue in this regard.

**Comment by alicera** [comment]
I try the Webmin 1.953 version.
And it is the same problem.
When I use the dynamic ip address setting, it doesnt have the Ethernet (Virtual).

**Comment by iliarostovtsev** [comment]
> When I use the dynamic ip address setting, it doesnt have the Ethernet (Virtual).

Excuse me, can you provide more details on how that can be reproduced?

**Comment by alicera** [comment]
ubuntu 18.04
sudo vim /etc/netplan/eno1.ymal
using the setting
"""
network:
  version: 2
  renderer: NetworkManager
  ethernets:
     eno1:
       dhcp4: true
"""

then, webmin will see the eno1
the Q1:  when I add the virtual interface with webmin,it wont see   the Ethernet (Virtual)
![image](https://user-images.githubusercontent.com/59388989/88743982-ac95ca80-d178-11ea-853f-40a138eac022.png)

Q2 : when I use the /etc/network/interface setting ,it wont see any Ethernet foe webmin.
But ubuntu16.04 can see it

**Comment by iliarostovtsev** [comment]
> But ubuntu16.04 can see it

_Netplan.io_ started to be used (unfortunately) starting with Ubuntu 17 only.

> the Q1: when I add the virtual interface with webmin,it wont see the Ethernet (Virtual)

I don't think it would be straight forward to display properly configuration _DHCP_ + _Static IP_. We would need to extract currently used IP acquired from DHCP and then merge it with data derived from `/etc/netplan/*.yaml`. I will discuss it with Jamie if it's worth a try. If you want correct representation of list of IPs on _Activated on Boot_ tab, you must not use _DHCP_ and switch to using _Static IP_ configuration instead.

The other great solution would be is to purge pointless _netplan.io_ package completely and install and configure _ifupdown_ to fallback to old good network configuration practices. :slightly_smiling_face:

**Comment by iliarostovtsev** [comment]
> I don't think it would be straight forward to display properly configuration DHCP + Static IP.

It was straight forward - I was mislead by _Active Now_ tab output, which is swapping IP addresses on Ubuntu 18.

**Comment by alicera** [comment]
So , webmin only can be used in Ubuntu16.04?

**Comment by iliarostovtsev** [comment]
> So , webmin only can be used in Ubuntu16.04?

No, no. :slightly_smiling_face: Ubuntu 18 and 20 are also already support. The bug above was already fixed.

**Comment by alicera** [comment]
ubuntu 18.04
sudo vim /etc/netplan/eno1.ymal
using the setting
"""
network:
version: 2
renderer: NetworkManager
ethernets:
eno1:
dhcp4: true
"""
![image](https://user-images.githubusercontent.com/59388989/89195077-cf8e0780-d5da-11ea-9210-366428c17347.png)

Do you try to add the virtual Ethernet for dynamic setting？？
I cant see the eno1：0
https://user-images.githubusercontent.com/59388989/88743982-ac95ca80-d178-11ea-853f-40a138eac022.png

**Comment by iliarostovtsev** [comment]
What is the output of:


~~~
ip a
~~~

**Comment by alicera** [comment]
I think you follow the steps and you will meet the same problem
step 1
sudo vim /etc/netplan/eno1.ymal
using the setting
"""
network:
version: 2
renderer: NetworkManager
ethernets:
eno1:
dhcp4: true
"""
then  "netplan apply "

step2
webmin setting
![image](https://user-images.githubusercontent.com/59388989/89245921-879fcc80-d63c-11ea-81b4-400cfa9f9d30.png)

step3
click " Add virtual interface" and set network
![image](https://user-images.githubusercontent.com/59388989/89246068-e2d1bf00-d63c-11ea-9f58-a63d36692eb8.png)

step4 
back to network Interfaces, you will see nothing.
![image](https://user-images.githubusercontent.com/59388989/89245921-879fcc80-d63c-11ea-81b4-400cfa9f9d30.png)

step5 
I expect the result eno1:0
![image](https://user-images.githubusercontent.com/59388989/88743982-ac95ca80-d178-11ea-853f-40a138eac022.png)

**Comment by iliarostovtsev** [comment]
Okay, what is the output of the following command on your Ubuntu 18.04, where you're having this issue:


~~~
ip --json addr
~~~

**Comment by alicera** [comment]
Option "-json" is unknown, try "ip -help".

what do you want?

**Comment by jcameron** [comment]
Also, just `ip addr` would work.
